File: Source/Seasonal/Shop/ShopChanger.cs
using WCSharp.Api;
using WCSharp.Shared.Extensions;
using static WCSharp.Api.Common;

public static class ShopChanger
{
    public static void Initialize() => Apply(SeasonalManager.CurrentTheme);

    /// <summary>Re-skins every vendor to match the given theme. Replaces the old season switch.</summary>
    public static void Apply(SeasonTheme theme)
    {
        var tempGroup = group.Create();
        tempGroup.EnumUnitsInRect(Globals.WORLD_BOUNDS, Filter(() => GetFilterUnit().UnitType == Constants.UNIT_KITTY_VENDOR));
        while (true)
        {
            var vendor = tempGroup.First;
            if (vendor == null) break;
            tempGroup.Remove(vendor);
            vendor.Skin = theme.VendorSkin;
        }
        GC.RemoveGroup(ref tempGroup);
    }

    // Compatibility wrapper - remove once you've confirmed nothing outside
    // these files calls it directly.
    public static void SetSeasonalShop() => Apply(SeasonalManager.CurrentTheme);
}

File: src/Seasonal/Shop/ShopChanger.ts
export class ShopChanger {
    private static SantaSkin: number = Constants.UNIT_SANTA
    private static HalloweenSkin: number
    private static ValentinesSkin: number
    private static EasterSkin: number
    private static NormalSkin: number = Constants.UNIT_KITTY_VENDOR

    public static Initialize() {
        if (SeasonalManager.Season == HolidaySeasons.None) return
        SetSeasonalShop()
    }

    public static SetSeasonalShop() {
        switch (SeasonalManager.Season) {
            case HolidaySeasons.None:
                SetShopsToSkin(NormalSkin)
                break

            case HolidaySeasons.Christmas:
                SetShopsToSkin(SantaSkin)
                break
            /*            case HolidaySeasons.Halloween:
                                HalloweenShop();
                                break;

                            case HolidaySeasons.Easter:
                                EasterShop();
                                break;

                            case HolidaySeasons.Valentines:
                                ValentinesShop();
                                break;*/
        }
    }

    private static SetShopsToSkin(skinType: number) {
        let tempGroup = Group.create()!
        tempGroup.EnumUnitsInRect(
            Globals.WORLD_BOUNDS,
            Filter(() => getFilterUnit().typeId == Constants.UNIT_KITTY_VENDOR)
        )
        while (true) {
            let vendor = tempGroup.First
            if (vendor == null) break
            tempGroup.Remove(vendor)
            vendor.skin = skinType
        }
        GC.RemoveGroup(tempGroup) // TODO; Cleanup:         GC.RemoveGroup(ref tempGroup);
    }
}
